internal/collectors/cloudprovider.go

```diff
@@ -0,0 +1,47 @@
+package collectors
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/SUSE/connect-ng/internal/util"
+)
+
+var (
+	cloudEx = `Version: .*(amazon)|Manufacturer: (Amazon)|Manufacturer: (Google)|Manufacturer: (Microsoft) Corporation`
+	cloudRe = regexp.MustCompile(cloudEx)
+)
+
+// findCloudProvider returns the cloud provider from "dmidecode -t system" output
+func findCloudProvider(b []byte) (string, bool) {
```

Comment:
Can't we just assume that if an empty string is returned, then no cloud provider was detected? Maybe we can add a debug line to be extra sure, but returning a "found" bool is kind of weird in golang I think.

Reply:
Is it, I thought the:

```
if x, ok := action(); ok {

}
```
semantic is pretty common, no?

Reply:
Ok, and there's also the part of not sending empty strings, and this would ease up the implementation. Nevermind, then 👍🏻